cookbooks/bcpc/recipes/default.rb

```diff
@@ -19,32 +19,55 @@
 
 require 'ipaddr'
 
-mgmt_cidr = IPAddr.new(node['bcpc']['management']['cidr'])
+ifs = node[:network][:interfaces].keys
 
-ifs=node[:network][:interfaces].keys
 # create a hash of ipaddresses -- skip interfaces without addresses
-ips= ifs.map{ |a| node[:network][:interfaces][a].attribute?(:addresses) and
+ips = ifs.map{ |a| node[:network][:interfaces][a].attribute?(:addresses) and
                      node[:network][:interfaces][a][:addresses] or {}}.reduce({}, :merge)
 
+# build a list of networks on this machine
+nets = ips.keys.select{ |ip| ips[ip]['family'] == "inet" }.map{ |ip| IPAddr.new("#{ip}/#{ips[ip]['prefixlen']}") }
+
+# find which subnet contains this machine's management network
+subnet = node[:bcpc][:networks].keys.map do |env_net|
+  matches = nets.select{|n| n == IPAddr.new(node[:bcpc][:networks][env_net][:management][:cidr])}
```

Comment:
Hrm, per [IPAddr.include?](http://docs.ruby-lang.org/en/2.0.0/IPAddr.html#method-i-include-3F) should likely be `===` or more explicitly/dogmatically `include?`.